clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -2014,6 +2238,370 @@ test_synthetic_time_quanta()
+        check_next(sched_cpu1, cpu1, time, scheduler_t::STATUS_OK, TID_A,
+                   // Ensure indirect_branch_target is correctly set to the 2nd instr
+                   // in TID_A.
+                   TRACE_TYPE_INSTR_INDIRECT_JUMP, THREAD_SWITCH_PC_START + 1, NO_MARKER,
+                   instr_2_tid_a_pc);
+        check_next(sched_cpu1, cpu1, time, scheduler_t::STATUS_OK, TID_A,
+                   TRACE_TYPE_MARKER,
+                   scheduler_tmpl_t<memref_t, reader_t>::switch_type_t::SWITCH_THREAD,
+                   TRACE_MARKER_TYPE_CONTEXT_SWITCH_END);
+
+        check_next(sched_cpu1, cpu1, time, scheduler_t::STATUS_OK, TID_A,
+                   TRACE_TYPE_INSTR, instr_2_tid_a_pc);
+        check_next(sched_cpu1, cpu1, time, scheduler_t::STATUS_OK, TID_A,
+                   TRACE_TYPE_MARKER, 0, TRACE_MARKER_TYPE_TIMESTAMP);
+        check_next(sched_cpu1, cpu1, time, scheduler_t::STATUS_OK, TID_A,
+                   TRACE_TYPE_MARKER, SYSCALL_BASE, TRACE_MARKER_TYPE_SYSCALL);
+        check_next(sched_cpu1, cpu1, time, scheduler_t::STATUS_OK, TID_A,
+                   TRACE_TYPE_MARKER, 0, TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL);
+
+        // Injected syscall sequence after the syscall instr in TID_A.
+        check_next(sched_cpu1, cpu1, ++time, scheduler_t::STATUS_OK, TID_A,
+                   TRACE_TYPE_MARKER, SYSCALL_BASE,
+                   TRACE_MARKER_TYPE_SYSCALL_TRACE_START);
+        check_next(sched_cpu1, cpu1, time, scheduler_t::STATUS_OK, TID_A,
+                   TRACE_TYPE_INSTR, SYSCALL_PC_START);
+        // We just hit a blocking syscall in A and there is nothing else to run.
+        // The indirect_branch_target of the last instr of the syscall sequence
+        // is not yet determinable, so we just see idles.
+        check_next(sched_cpu1, cpu1, ++time, scheduler_t::STATUS_IDLE);
+        // Finish off C on cpu 0.  This hits a quantum end but there's no one else.
+        check_next(sched_cpu0, cpu0, ++time, scheduler_t::STATUS_OK, TID_C,
+                   TRACE_TYPE_INSTR);
+        check_next(sched_cpu0, cpu0, ++time, scheduler_t::STATUS_OK, TID_C,
+                   TRACE_TYPE_INSTR);
+        check_next(sched_cpu0, cpu0, time, scheduler_t::STATUS_OK, TID_C,
+                   TRACE_TYPE_THREAD_EXIT);
+        // Both cpus wait until A is unblocked.
+        check_next(sched_cpu1, cpu1, ++time, scheduler_t::STATUS_IDLE);
+        check_next(sched_cpu0, cpu0, ++time, scheduler_t::STATUS_IDLE);
+        check_next(sched_cpu1, cpu1, ++time, scheduler_t::STATUS_IDLE);
+        check_next(sched_cpu0, cpu0, ++time, scheduler_t::STATUS_IDLE);
+        check_next(sched_cpu1, cpu1, ++time, scheduler_t::STATUS_IDLE);
+        check_next(sched_cpu0, cpu0, ++time, scheduler_t::STATUS_IDLE);
+        check_next(sched_cpu1, cpu1, ++time, scheduler_t::STATUS_IDLE);
+
+        // Now the indirect_branch_target of the last syscall sequence instr is
```

Comment:
How do you know A is unblocked here? The value of time? Add a comment on what value time has now, maybe an assert on it?

Reply:
Same as test_synthetic_time_quanta, the input is blocked due to the blocking syscall, the prior check_next returns STATUS_IDLE, and when the value of time for the next call is 19, it gets the next instr. Added assert on time.